src/window/building/distribution.c

```diff
@@ -151,6 +170,44 @@ static void draw_granary_permissions_buttons(int x, int y, int buttons)
     }
 }
 
+static void init_dock_permission_buttons()
+{
+  dock_distribution_permissions_buttons_count = 0;
+  for (int route_id = 0; route_id < 20; route_id++) {
+    int city_id = -1;
+    if (is_sea_trade_route(route_id) && empire_city_is_trade_route_open(route_id)) {
+      city_id = empire_city_get_for_trade_route(route_id);
+      if (city_id != -1) {
+        generic_button button = {0, 0, 210, 22, dock_toggle_route, button_none, route_id, city_id};
+        dock_distribution_permissions_buttons[dock_distribution_permissions_buttons_count] = button;
+        dock_distribution_permissions_buttons_count++;
+      }
+    }
+  }
+}
+
+static void draw_dock_permission_buttons(int x_offset, int y_offset, int dock_id)
+{
+  int button_order = 0;
```

Comment:
The "there are no open sea trade routes" text would be added here if `dock_distribution_permissions_buttons_count == 0`

Reply:
done 
![image](https://user-images.githubusercontent.com/1790512/106454260-25147580-6493-11eb-8086-ddf616f40989.png)
also it consider the height 
![image](https://user-images.githubusercontent.com/1790512/106454347-45443480-6493-11eb-8d08-7d1764d9b790.png)